price_feeds_poller.py

```diff
@@ -0,0 +1,387 @@
+        "gas": network_gas,
+        "gasPrice": network_gas_price,
+        "value": fee
+      })
+      log_master_balance(csv_filename, network_from, balance, tx.hex())
+
+      # Get receipt of the transaction
+      print(f" - Witnet    : {contract.functions.witnet().call()}")
+      print(f" - Pricefeed : {contract.address}")
+      print(f" - Account   : {network_from}")
+      print(f" - Balance   : {round(balance / 10 ** 18, 5)} {network_symbol}")
+      print(f" - Paying fee: {round(fee / 10 ** 18, 5)} {network_symbol}")
+      print( " - Gas limit :", "{:,}".format(network_gas))
+      print( " - Gas price :", "{:,}".format(network_gas_price))
+      print(f" = Tx hash   : {tx.hex()}")
+      receipt = w3.eth.waitForTransactionReceipt(
+        tx,
+        network_tx_waiting_timeout_secs,
+        network_tx_polling_latency_secs
+      )
+      print( " = Tx.eff.gas:", "{:,}".format(receipt.get("gasUsed")))
+      print( " = Total cost:", round((balance - w3.eth.getBalance(network_from)) / 10 ** 18, 5), network_symbol)
+
+    except exceptions.TimeExhausted:
+      print(f" * Transaction is taking too long.")
+      return 0
+
+    except Exception as ex:
+      print(f" x Transaction rejected: {ex}")
+      return 0
+
+    # Check if transaction was succesful
+    if receipt['status'] == False:
+      print(f" x Transaction reverted!")
+      return 0
+    else:      
+      requestId = contract.functions.latestQueryId().call()
+      print(f" < Request id: {requestId}")
+      return requestId
+
+def log_master_balance(csv_filename, addr, balance, txhash):
+  if csv_filename is not None:
+    try:
+      with open(csv_filename, "a", encoding="utf-8") as csv_file:
+        readable_ts = datetime.datetime.fromtimestamp(int(time.time())).strftime('%Y-%m-%d %H:%M:%S %Z')
+        row = f"\"{os.path.splitext(os.path.basename(csv_filename))[0]}\";\"{addr}\";\"{readable_ts}\";\"{balance}\";\"{txhash}\""
+        csv_file.write(row + '\n')
+    except:
+      return
+
+def log_exception_state(addr, reason):
+  # log the error and wait 1 second before next iteration
+  print(f"Exception while getting state from contract {addr}:\n{reason}")
+  time.sleep(1)
+
+@contextlib.contextmanager
+def stdoutIO(stdout=None):
+  old = sys.stdout
+  if stdout is None:
+    stdout = StringIO()
+  sys.stdout = stdout
+  yield stdout
+  sys.stdout = old
+
+def dry_run_request(bytecode, timeout_secs):
+  cmdline = "npx witnet-toolkit try-data-request --hex "
+  cmdline += bytecode.hex()
+  cmdline += " | tail -n 2 | head -n 1 | awk -F: '{ print $2 }' | sed 's/ //g' | tr -d \"│\""
+  
+  # Dry-run result needs to be fetched from temporary file, 
+  # because of https://bugs.python.org/issue30154.
+  with open("tmp.out", "w+") as output:
+    process = subprocess.Popen(
+      cmdline,
+      stdout = output,
+      shell = True,
+    )
+    timer = Timer(timeout_secs, process.kill)
+    try:
+      timer.start()
+      process.wait()
```

Comment:
Why use a timer instead of the `timeout` argument?

```suggestion
      process.wait(timeout=timeout_secs)
```

https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait